cli/src/semgrep/state.py

```diff
@@ -25,6 +35,92 @@ class SemgrepState:
     settings: Settings = Factory(Settings)
     terminal: Terminal = Factory(Terminal)
 
+    @staticmethod
+    def get_cli_ux_flavor() -> DesignTreatment:
+        """
+        Returns the CLI UX flavor to use for the current CLI invocation.
+        """
+        new_cli_ux = get_state().env.with_new_cli_ux
+        if not new_cli_ux:
+            return DesignTreatment.LEGACY
+        config = get_config()
+        # NOTE: We only support simple and detailed UX treatments for `semgrep scan` invocations
+        if SemgrepState.is_scan_invocation():
+            # NOTE: Ignore the default 'auto' config and product flags such as 'supply-chain'
+            has_config = bool(set(config) - {"auto", "supply-chain"})
+            return DesignTreatment.DETAILED if has_config else DesignTreatment.SIMPLE
+        return DesignTreatment.DETAILED
+
+    @staticmethod
+    def is_legacy_cli_ux() -> bool:
```

Comment:
I would remove that and instead pass around the UXStyle and just use plain comparison each time.

Reply:
I hear you – and this goes back to an earlier conversation  on whether we need / should use simple helper functions when a direct comparison will do. My reason for this simple helper was to standardize checking stateful properties (e.g. design treatment and supply-chain)